lang/cs/Org.Apache.REEF.Common/Telemetry/MetricData.cs

```diff
@@ -0,0 +1,140 @@
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    /// <summary>
+    /// This class wraps a metric object and the increment value since last sink
+    /// </summary>
+    [JsonObject]
+    public sealed class MetricData
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(MetricsData));
+
+        /// <summary>
+        /// Metric object
+        /// </summary>
+        [JsonProperty]
+        private IMetric _metric;
+
+        [JsonProperty]
+        private IList<IMetric> _records;
+
+        ///// <summary>
+        ///// Whether metric has been updated since last sink.
+        ///// </summary>
+        [JsonProperty]
+        internal int ChangesSinceLastSink;
+
+        /// <summary>
+        /// Constructor for metricData
+        /// </summary>
+        /// <param name="metric"></param>
+        /// <param name="initialValue"></param>
+        internal MetricData(IMetric metric)
+        {
+            _metric = metric;
+            ChangesSinceLastSink = 0;
+            _records = new List<IMetric>();
+        }
+
+        [JsonConstructor]
+        internal MetricData(IMetric metric, IList<IMetric> records, int changes)
+        {
+            _metric = metric;
+            _records = records;
+            ChangesSinceLastSink = changes;
+        }
+
+        /// <summary>
+        /// clear the increment since last sink
+        /// </summary>
+        internal void ResetChangeSinceLastSink()
+        {
+            ChangesSinceLastSink = 0;
+            _records.Clear();
+        }
+
+        /// <summary>
+        /// When new metric data is received, update the value and records so it reflects the new data.
+        /// </summary>
+        /// <param name="metric">Metric data received.</param>
+        internal void UpdateMetric(MetricData metric)
+        {
+            if (!(metric.GetMetric() is ICounter))
+            {
+                foreach (var r in metric._records)
+                {
+                    _records.Add(r);
+                }
+                _records.Add(_metric);
+            }
+            ChangesSinceLastSink += metric.ChangesSinceLastSink;
+            _metric = metric.GetMetric(); // update current metric value
+        }
+
+        internal void UpdateMetric(IMetric me)
```

Comment:
This method seems to share a lot of code with the other UpdateMetric method.

Reply:
added private UpdateRecords